src/apps/request_index_app.tsx

```diff
@@ -8,19 +8,26 @@ import Divider from "../components/divider";
 import Table from "../components/table";
 import RequestService from "../services/request_service";
 import IncludeChildren from "../components/include_children_checkbox";
+import CacheService from "../services/cache_service";
+import MyState from "../custom_types/table_state_type";
+
+type CachedStateType = {
+  rowsPerPage: number;
+};
 
 class RequestApp extends Component {
-  state = {
-    response: {},
+  cachedState: CachedStateType = CacheService.getIndexLastState(
```

Comment:
I'm still somewhat concerned about the amount of duplication in all of these components.  Setting up this cachedState and the updateData function in particular, but really just in general. We should continue thinking about ways to abstract out / centralize a lot of this so that we're duplicating as little as possible.

Reply:
Some of the duplication stuff should be resolved in the work I am doing for issue #34 since I am needing to relocate some things to not be using any.